packages/opentelemetry-plugin-http/src/http.ts

```diff
@@ -295,10 +308,23 @@ export class HttpPlugin extends BasePlugin<Http> {
       };
 
       return context.with(propagation.extract(headers), () => {
-        const span = plugin._startHttpSpan(
-          `${method} ${pathname}`,
-          spanOptions
-        );
+        let span: Span = new NoopSpan();
+        const hasParent = plugin._tracer.getCurrentSpan() !== undefined;
+        /*
+         * If a parent is required but not present, we use a `NoRecordingSpan` to still
+         * propagate context without recording it.
+         */
+        if (
+          plugin._config.requireParentforIncomingSpans === true &&
+          hasParent === false
+        ) {
+          const spanContext =
+            getExtractedSpanContext(context.active()) ??
+            plugin._emptySpanContext;
+          span = new NoRecordingSpan(spanContext);
```

Comment:
@mayurkale22 WDYT about this? According to spec, spans are not supposed to be created directly.

Reply:
friendly up @mayurkale22

Reply:
Sorry I missed this. I agree this is not compliant with the spec. Maybe, for now add `@todo` here and open issue to address this later.